civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -5929,12 +5929,17 @@ function civihr_employee_portal_user_login_submit(&$form, &$form_state) {
 }
 
 /**
- * Implements hook_user_login
+ * Implements hook_user_login().
  */
 function civihr_employee_portal_user_login(&$edit, $account) {
-  global $base_url;
+
   if (!isset($_POST['form_id']) || $_POST['form_id'] != 'user_pass_reset') {
-    $_GET['destination'] = user_access('access CiviCRM') ?  'civicrm/tasksassignments/dashboard#/tasks' : 'dashboard';
+    if (current_path() != drupal_get_destination()['destination']) {
```

Comment:
This can be rewritten without the else and with less duplication:
```php
$sspEntryPath = 'dashboard';
$civicrmEntryPath = 'civicrm/tasksassignments/dashboard#/tasks';
if (current_path() != drupal_get_destination()['destination']) {
   $civicrmEntryPath = $_GET['destination'];
}

$_GET['destination'] = user_access('access CiviCRM') ? $civicrmEntryPath : $sspEntryPath;
```